Assignment 1/script.js

```diff
@@ -1 +1,191 @@
+        endDateInput.min = startDateInput.value;
+        addWeekPreset.classList.remove("isDisabled");
+        addMonthPreset.classList.remove("isDisabled");
+    }
+    else{
+        endDateInput.disabled = true;
+        addWeekPreset.classList.add("isDisabled");
+        addMonthPreset.classList.add("isDisabled");
+    }
+}
+
+function trySubmitForm (event) {
+    event.preventDefault();
+
+    if (form.checkValidity()) {
+        const startDate = startDateInput.value;
+        const endDate = endDateInput.value;
+        const result = getDuration(
+            startDate,
+            endDate,
+            getRadioValue(timeRadios),
+            getRadioValue(daysRadios)
+        );
+
+        addNewRequestToLocalStorage(startDate, endDate, result);
+    }
+}
+
+function dateFromPresset(date, preset = "week"){
+    let newDate;
+    switch (preset){
+        case "month":
+            newDate = new Date(new Date(date).setMonth(new Date(date).getMonth() + 1))
+            break;
+        case "week":
+            newDate = new Date(new Date(date).setDate(new Date(date).getDate() + 7))
+            break;
+    }
+    return newDate.getFullYear() + "-" + String(newDate.getMonth() + 1).padStart(2, '0') + "-" + String(newDate.getDate()).padStart(2, '0');
+}
+
+function convertDate(date) {
+    date = new Date(Date.parse(date));
+
+    return String(date.getDate()).padStart(2, '0') + "/" +
+    String(date.getMonth() + 1).padStart(2, '0') + "/" +
+    date.getFullYear();
+}
+
+function getDuration(startDate, endDate, units = "days", type = "all days"){
+    function convertDaysInto(time) {
+        switch (units) {
+            case "days":
+                time = time;
+                break;
+            case "hours":
+                time = time*24;
+                break;
+            case "minutes":
+                time = time*60*24;
+                break;
+            case "seconds":
+                time = time*3600*24;
+                break;
+        }
+        return time;
+    }
+
+    const durationInDays = getDurationInDays(endDate, startDate);
+    let result;
+    
+    switch (type) {
+        case "all days":
+            result = convertDaysInto(durationInDays);
+            break;
+        case "working days":
+            result = convertDaysInto(durationInDays - getWeekendDays(startDate, endDate)) + " working";
+            break;
+        case "weekend days":
+            result = convertDaysInto(getWeekendDays(startDate, endDate)) + " weekend"
+            break;
+    }
+    result += " " + units;
+    return result;
+}
+
+function getDurationInDays(endDate, startDate) {
+    return (Date.parse(endDate) - Date.parse(startDate))/1000/3600/24;
+}
+
+function getWeekendDays(startDate, endDate){
+    let days = getDurationInDays(endDate, startDate);
+    let result = 0;
+    startDate = new Date(Date.parse(startDate)).getDay();
+    endDate = new Date(Date.parse(endDate)).getDay();
+
+    if (startDate === 0) result += 1;
```

Comment:
```suggestion
    if (startDate === 0) {
        result += 1
    };
```

Не треба економити на дужках)